### ISSUE TITLE
Bootstrapping issue

### ISSUE THREAD

**Issue by klaascuvelier** [AntJanus/angular2-ts-starter-kit#3]
Ran into another issue when using this project.
When opening the page, I noticed this error in the console.

```
EXCEPTION: The selector "my-app" did not match any elements
bundle.js:433 Uncaught (in promise) TypeError: object is not a constructor(…)obj.(anonymous function) @ bundle.js:433
bundle.js:23282 Uncaught (in promise) BaseException {message: "The selector "my-app" did not match any elements", stack: "Error: The selector "my-app" did not match any ele…eBoundFn (http://localhost:3001/bundle.js:177:20)"}(anonymous function) @ bundle.js:23282run @ bundle.js:204(anonymous function) @ bundle.js:23697zoneBoundFn @ bundle.js:177
```

Took me some time to figure out what the issue was, but basically, the `bootstrap` function was called to early.
Not sure why you did not run into this issue. Maybe something recently changed?

Anyway, there are 2 ways to solve this:
- move the script tag to the body bottom
- add an event listener for DOMContentLoaded

I'll create a PR for both options, I'll let you decide what you think is the best option.
(I'd prefer option 2)

**Comment by AntJanus** [comment]
Yeah, I never ran into the issue. In fact, I have an app out in the wild using this starter kit and it works on the same principle as this starter kit (I used the starter kit for this).

Can you give me some details on what you were doing? Was it just the starter kit without any changes? What browser/OS was this on?

I'll try to troubleshoot it.

**Comment by klaascuvelier** [comment]
I am using Chrome (Canary: `Version 49.0.2591.0`)
I just cloned the repo, did the `npm install` (see different issue) and ran `gulp`.

For the record, I checked it in Firefox and Safari, and it does seem to work in there without any changes.
But it confuses me, as maybe it shouldn't work. The script is in the head, and gets executed before the DOM body is ready. 
I did some debugging in the console:
`var element = dom_adapter_1.DOM.querySelector(this._document, hostElementSelector);`
`element` is null, as the body is not ready, as a script tag is blocking, and the DOM element is after the script tag ...

**Comment by AntJanus** [comment]
Strange. Angular 2 should have a built-in wait for bootstrapping.  If you check the [Angular quickstart](https://angular.io/docs/ts/latest/quickstart.html) and scroll to the bottom, you'll see that they're not waiting for DOM either. 

Let me ask you something, if you go through [this article](http://antjanus.com/blog/tutorials/the-beginners-preemptive-guide-to-angularjs-2-alpha/) there should be iframes to an example app. Does it work for you? Also, do you have any extensions that might prevent any javascript from loading?

**Comment by klaascuvelier** [comment]
Correct. I noticed that as well. AngularClass however does use the listener:
https://github.com/AngularClass/angular2-webpack-starter/blob/master/src/bootstrap.ts#L29
(And some others are as well, that is how I found this issue)

On the examples, the first 2 aren't loading (in Chrome), but I think the reason might be unrelated? 
`bundle.js:7605 Uncaught TypeError: RegExp.prototype.global getter called on non-RegExp object`

The last was is loading, but is acting weird (render glitches). Maybe all of this is related to the fact that I am using Chrome Canary (49.0.2591.0 canary (64-bit))

On the other hand, I think it's "nicer" to add the event listener anyway ...
